src/lightning/pytorch/trainer/connectors/logger_connector/result.py

```diff
@@ -206,13 +206,15 @@ def update(self, value: _VALUE, batch_size: int) -> None:
         if self.is_tensor:
             value = cast(Tensor, value)
             if not torch.is_floating_point(value):
-                dtype = torch.get_default_dtype()
+                dtype = _get_default_dtype()
                 warning_cache.warn(
                     # do not include the value to avoid cache misses
                     f"You called `self.log({self.meta.name!r}, ...)` in your `{self.meta.fx}` but the value needs to"
                     f" be floating point. Converting it to {dtype}."
                 )
                 value = value.to(dtype)
+            if value.dtype in (torch.float16, torch.bfloat16):
```

Comment:
Using "not in" would future proof this for newer dtypes (8bit etc)

Reply:
agreed, done.